python_modules/dagster/dagster/_utils/pydantic_yaml.py

```diff
@@ -1,15 +1,68 @@
-from typing import Type, TypeVar
+from typing import List, Optional, Type, TypeVar
 
 from pydantic import BaseModel, ValidationError, parse_obj_as
 
+from dagster._core.errors import DagsterInvariantViolationError
 from dagster._model.pydantic_compat_layer import USING_PYDANTIC_1
 
-from .source_position import KeyPath, populate_source_position_and_key_paths
+from .source_position import (
+    KeyPath,
+    ValueAndSourcePositionTree,
+    populate_source_position_and_key_paths,
+)
 from .yaml_utils import parse_yaml_with_source_positions
 
 T = TypeVar("T", bound=BaseModel)
 
 
+def _parse_and_populate_model_with_annotated_errors(
+    cls: Type[T],
+    obj_parse_root: ValueAndSourcePositionTree,
+    file_root: Optional[ValueAndSourcePositionTree],
```

Comment:
Trying to wrap my mind around this - why exactly isn't `obj_parse_root` enough? Does it not know its position within the file?

Reply:
Ah yeah you're right, dropped it

Reply:
It's still there, no?

Reply:
ah yeah I think my push didn't work properly, fixed